backend/app/api/endpoints/adapter/attachments.py

```
@@ -271,13 +276,33 @@ def _validate_share_token_access(
+    return is_pdf and file_size > size_threshold
+
+
 @router.post("/upload", response_model=AttachmentResponse)
 async def upload_attachment(
     file: UploadFile = File(...),
     overwrite_attachment_id: Optional[int] = None,
+    parse_async: bool = Query(
+        default=False,
+        description="Parse document asynchronously. PDF files larger than 1MB will be parsed in background.",
+    ),
     db: Session = Depends(get_db),
```

Ruff (0.15.9)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)